src/memory_manager.rs

```diff
@@ -454,7 +454,7 @@ pub fn gc_poll<VM: VMBinding>(mmtk: &MMTK<VM>, tls: VMMutatorThread) {
         "gc_poll() can only be called by a mutator thread."
     );
 
-    if mmtk.state.should_trigger_gc_when_heap_is_full() && mmtk.gc_trigger.poll(false, None) {
+    if !(VM::VMCollection::is_collection_disabled()) && mmtk.gc_trigger.poll(false, None) {
```

Comment:
`should_trigger_gc_when_heap_is_full` may be unused for this change, and can be removed?